src/member/member.ts

```diff
@@ -52,22 +53,53 @@ export const buildMemberExtra = (extra: Partial<MemberExtra>) => ({
   ...extra,
 });
 
-export type Member = {
+export type Account = {
   id: UUID;
   name: string;
-  email: string;
 };
 
-export type CompleteMember = Member & {
+type AccountTypeProperty = {
   type: `${MemberType}` | MemberType;
+};
+
+export type Member = Account & {
+  email: string;
+};
+
+export type Guest = Account;
+
+export type AugmentedAccount =
+  | (Member & {
+      type: MemberType.Individual;
+    })
+  | (Guest & {
+      type: MemberType.Guest;
+    });
+
+export type BaseAccount = Account &
+  AccountTypeProperty & {
+    createdAt: string;
+    updatedAt: string;
+    lastAuthenticatedAt?: string;
+  };
+
+export type CompleteMember = BaseAccount & {
+  type: MemberType.Individual;
+  email: string;
   extra: MemberExtra;
   enableSaveActions: boolean;
   userAgreementsDate?: string;
-  createdAt: string;
-  updatedAt: string;
-  lastAuthenticatedAt?: string;
+
   isValidated: boolean;
 };
 
-export const isPseudoMember = (member: { email: string }) =>
-  PSEUDO_USER_MAIL_REGEX.test(member.email.toLowerCase());
+export type CompleteGuest = BaseAccount & {
+  type: MemberType.Guest;
+  itemLoginSchema: ItemLoginSchema;
+};
+
+export type CompleteAccount = CompleteMember | CompleteGuest;
```

Comment:
Also here `CompleteAccount` didn't really reflects the union of types (CompleteMember and CompleteGuest). 😅

Again I don't really have a good suggestion for another name ^^